### ISSUE TITLE
Documentation [Feature Request]

### ISSUE THREAD

**Issue by TACIXAT** [KhronosGroup/SPIRV-Reflect#294]
Anything that documents the API - doxygen, a tutorial. I'm working with Vulkan and have decided to go the Slang route and this library seems pretty essential to that ecosystem. I can get by with LLMs for now but API docs would be rad. Thank you!

**Comment by spencer-lunarg** [comment]
I agree, I have wanted to add a "how to use SPIRV-Reflect" for a while

**Comment by FuXiii** [comment]
Better apply a document like `Vulkan doc` style. Thanks. •ᴗ•
